src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.wasdev.wlp.common.plugins.util;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.Watchable;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+import org.apache.commons.io.FileUtils;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+
+/**
+ * Utility class for dev mode.
+ */
+public abstract class DevUtil {
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     */
+    public abstract void debug(String msg);
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     * @param e
+     */
+    public abstract void debug(String msg, Throwable e);
+
+    /**
+     * Log debug
+     * 
+     * @param e
+     */
+    public abstract void debug(Throwable e);
+
+    /**
+     * Log warning
+     * 
+     * @param msg
+     */
+    public abstract void warn(String msg);
+
+    /**
+     * Log info
+     * 
+     * @param msg
+     */
+    public abstract void info(String msg);
+
+    /**
+     * Log error
+     * 
+     * @param msg
+     */
+    public abstract void error(String msg);
+
+    /**
+     * Returns whether debug is enabled by the current logger
+     * 
+     * @return whether debug is enabled
+     */
+    public abstract boolean isDebugEnabled();
+
+    /**
+     * Stop the server
+     */
+    public abstract void stopServer();
+
+    /**
+     * Starts the default server
+     */
+    public abstract void startServer();
+
+    /**
+     * Updates artifacts of current project
+     */
+    public abstract void getArtifacts(List<String> artifactPaths);
```

Comment:
It seems awkward to get passed in a list and then add to it in this method. Why not just return a `List<String>` instead?

Reply:
Why was this issue marked resolved without comment?

Reply:
it was changed as suggested

Reply:
If it was changed wouldn't this comment get marked as outdated? I am still seeing it returning a `void` instead of `List<String>`.

Reply:
Not sure why.  The updated version can be seen here https://github.com/WASdev/ci.common/pull/48/files#diff-40ef4b1443d5315fd48803cbbcb671b0R137